### PR TITLE
upgrade URLs to HTTPS (1 of 2)

### PR DESCRIPTION
**Pull request by vszakats**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

It's been a few years that I run my old URL upgrader script on Homebrew core. Here are the ones it has found. Some I found manually and verified those that couldn't be auto-verified. Also skipped those that shouldn't be upgraded for one reason or another.

### CODE REVIEW

**Comment by SMillerDev** [comment]
Could you split every formula into their own commit please?

**Comment by vszakats** [comment]
This was the request that made me leave this project after doing similar upgrades for 5 years in batches of usually 10 (as agreed with maintainers here). It doesn't seem to make much sense now to split this into 100 PRs IMO.

Please take this as it is or just close it.

**Comment by bevanjkay** [comment]
We aren't asking for it to be split into 100 PRs, but one commit per formula. This makes the history much easier to track, and for changes to be reverted where required more easily.

**Comment by vszakats** [comment]
Oh okay, I'll be looking at this, though it will likely need a newly written script to finish with it in a reasonable time. Unless I find the one I used back then.

**Comment by bevanjkay** [comment]
I can link a script to commit each file individually. You'll just need to revert the commit and make sure the edits are applied. I can drop it here later.

**Comment by vszakats** [comment]
Thanks @bevanjkay, managed to split it to one commit per file.

**Comment by vszakats** [comment]
Quite a few failures, none seem to be related to these updates.
I will not be able to address anything unrelated to these, sorry.

**Comment by chenrui333** [comment]
I think it should syntax only for just checking the urls?

**Comment by vszakats** [comment]
> I think it should syntax only for just checking the urls?
> 
> relates to #174428

Yes.

3 commits upgrade URLs in tests, of which `go-camo` actually uses that URL. That one passed CI tests in earlier pushes.

**Comment by vszakats** [comment]
Is there something left to fix with this PR to have it merged?

File: Formula/lib/libgrapheme.rb
Comment (SMillerDev) [inline Formula/lib/libgrapheme.rb:7]: Why?

Comment (vszakats) [reply Formula/lib/libgrapheme.rb:7]: Because the git protocol is cleartext, unencrypted and unauthenticated. Like HTTP.

**Comment by vszakats** [comment]
Any more concern to address?

**Comment by vszakats** [comment]
One half of this effort has been merged weeks ago. What is holding back the other part? CI is green. (At least it was for weeks, we will now see after fixing a merge conflict that emerged in the meantime.)

Short of any concern, if there is no desire to make URLs secure, I'm happy to close this, though the fixes are valid and do improve security and confidentiality.

**Comment by vszakats** [comment]
Is there something to be done to make this merged?

**Review by p-linnane — APPROVED** [review]
Looks good. Thanks @vszakats!

**Comment by vszakats** [comment]
Thank you @p-linnane!

**Pull request merged**
